src/Microsoft.AspNetCore.Mvc.Core/ApplicationParts/ProvideApplicationPartFactoryAttribute.cs

```diff
@@ -11,15 +12,47 @@ namespace Microsoft.AspNetCore.Mvc.ApplicationParts
     [AttributeUsage(AttributeTargets.Assembly, AllowMultiple = false)]
     public sealed class ProvideApplicationPartFactoryAttribute : Attribute
     {
+        /// <summary>
+        /// Creates a new instance of <see cref="ProvideApplicationPartFactoryAttribute"/> with the specified type.
+        /// </summary>
+        /// <param name="factoryType">The factory type.</param>
         public ProvideApplicationPartFactoryAttribute(Type factoryType)
         {
             ApplicationPartFactoryType = factoryType ?? throw new ArgumentNullException(nameof(factoryType));
         }
 
+        /// <summary>
+        /// Creates a new instance of <see cref="ProvideApplicationPartFactoryAttribute"/> with the specified type name.
+        /// </summary>
+        /// <param name="factoryTypeName">The assembly qualified type name.</param>
         public ProvideApplicationPartFactoryAttribute(string factoryTypeName)
         {
+            if (string.IsNullOrEmpty(factoryTypeName))
+            {
+                throw new ArgumentException(Resources.ArgumentCannotBeNullOrEmpty, nameof(factoryTypeName));
+            }
+
+            ApplicationPartFactoryTypeName = factoryTypeName;
         }
 
-        public Type ApplicationPartFactoryType { get; }
+        /// <summary>
+        /// The factory type.
+        /// </summary>
+        private Type ApplicationPartFactoryType { get; }
```

Comment:
nit: Make them private readonly fields and remove the doc comments. It confused me for like 5 min to figure out what this was doing.